spring-vault-core/src/test/java/org/springframework/vault/core/lease/SecretLeaseContainerIntegrationTests.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.vault.core.lease;
+
+import java.util.HashMap;
+import java.util.Map;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.vault.core.VaultIntegrationTestConfiguration;
+import org.springframework.vault.core.VaultOperations;
+import org.springframework.vault.core.env.LeaseAwareVaultPropertySource;
+import org.springframework.vault.core.lease.RotatingGenericSecretsIntegrationTestConfiguration.PropertySourceHolder;
+import org.springframework.vault.support.VaultResponseSupport;
+import org.springframework.vault.util.PrepareVault;
+import org.springframework.vault.util.VaultRule;
+
+/**
+ * Integration tests for {@link SecretLeaseContainer} using the {@code generic} backend.
+ *
+ * @author Steven Swor
+ */
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = { VaultIntegrationTestConfiguration.class,
+		RotatingGenericSecretsIntegrationTestConfiguration.class })
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class SecretLeaseContainerIntegrationTests {
```

Comment:
Let's remove this integration test to keep the number of integration tests low. We have already unit tests for `SecretLeaseContainer` with renewal in `SecretLeaseContainerUnitTests` so the new test case could fit in there quite well.

Especially the timing aspect makes testing inconvenient because it adds a dependency on something we don't have under control.